featurebyte/query_graph/node/generic.py

```diff
@@ -1080,7 +1081,7 @@ def _derive_node_operation_info(
                     # so that any changes on the right_on column can be tracked.
                     right_columns[col.name] = DerivedDataColumn.create(
                         name=right_col_map[col.name],  # type: ignore
-                        columns=[right_on_col, col],
+                        columns=[right_on_col, col, left_on_col],
```

Comment:
This is the main fix, should include `left_on_col` as it contributes to the derivation of the column (as it affect the row index). Without including `left_on_col`, it causes issue if the `left_on_col` is related to the third table.